lib/ansible/modules/cloud/amazon/elb_target_group.py

```diff
@@ -0,0 +1,507 @@
+  name:
+    description:
+      - The name of the target group.
+    required: true
+  protocol:
+    description:
+      - The protocol to use for routing traffic to the targets. Required if state=present.
+    required: false
+    choices: [ 'http', 'https' ]
+  port:
+    description:
+      - The port on which the targets receive traffic. This port is used unless you specify a port override when registering the target. Required if state=present.
+    required: false
+  vpc_id:
+    description:
+      - The identifier of the virtual private cloud (VPC). Required if state=present.
+    required: false
+  health_check_protocol:
+    description:
+      - The protocol the load balancer uses when performing health checks on targets.
+    required: false
+    choices: [ 'http', 'https' ]
+  health_check_port:
+    description:
+      - The port the load balancer uses when performing health checks on targets.
+    required: false
+    default: "The port on which each target receives traffic from the load balancer."
+  health_check_path:
+    description:
+      - The ping path that is the destination on the targets for health checks. The path must be defined in order to set a health check.
+    required: false
+  health_check_interval:
+    description:
+      - The approximate amount of time, in seconds, between health checks of an individual target.
+    required: false
+    default: 30
+  health_check_timeout:
+    description:
+      - The amount of time, in seconds, during which no response from a target means a failed health check.
+    required: false
+    default: 5
+  healthy_threshold_count:
+    description:
+      - The number of consecutive health checks successes required before considering an unhealthy target healthy.
+    required: false
+    default: 5
+  unhealthy_threshold_count:
+    description:
+      - The number of consecutive health check failures required before considering a target unhealthy.
+    required: false
+    default: 2
+  successful_response_codes:
+    description:
+      - The HTTP codes to use when checking for a successful response from a target.
+    required: false
+    default: 200
+  state:
+    description:
+      - Create or destroy the target group.
+    required: true
+    choices: [ 'present', 'absent' ]
+extends_documentation_fragment:
+    - aws
+    - ec2
+notes:
+  - Once a target group has been created, only its health check can then be modified using subsequent calls
+'''
+
+EXAMPLES = '''
+# Note: These examples do not set authentication details, see the AWS Guide for details.
+
+# Create a target group with a default health check
+- elb_target_group:
+    name: mytargetgroup
+    protocol: HTTP
+    port: 80
+    vpc_id: vpc-01234567
+    state: present
+
+# Modify the target group with a custom health check
+- elb_target_group:
+    name: mytargetgroup
+    protocol: HTTP
+    port: 80
+    vpc_id: vpc-01234567
+    health_check_path: /
+    successful_response_codes: "200, 250-260"
```

Comment:
This definitely bears explaining more in the docs, since it's an odd format and hard for us to check clientside.